clash-lib/src/Clash/Netlist/Types.hs

```diff
@@ -192,6 +192,10 @@ data IdentifierSet
       -- ^ Identifier store
     } deriving (Generic, NFData, Show)
 
+-- This is an option flag to clean up names that may be valid identifiers
+-- but that particular tools e.g. Vivado may reject (tcl with '$' in ids).
+data SanitizeNames = Sanitize | NoSanitize deriving (Eq, Bounded, Read, Show)
```

Comment:
We don't need all these instances, In fact, I don't think we need any of these instances